frontend/src/pages/adminEmployeeEdit.js

```diff
@@ -0,0 +1,182 @@
+import { useSearchParams } from "react-router-dom";
+import MyAlert from "./components/myAlert";
+
+function AdminEmployeeEdit() {
+    const [searchParams] = useSearchParams();
+    const id = searchParams.get("id");
+
+    const [userName, setUserName] = useState();
+    const [password, setPassword] = useState();
+    const [managerName, setManagerName] = useState();
+    const [roleName, setRoleName] = useState();
+    const [annualLeaveEntitlement, setAnnualLeaveEntitlement] = useState();
+    const [medicalLeaveEntitlement, setMedicalLeaveEntitlement] = useState();
+    const [compensationLeaveEntitlement, setCompensationLeaveEntitlement] = useState();
+
+    const [showAlert, setShowAlert] = useState(false);
+    const [alertMsg, setAlertMsg] = useState();
+
+    const onInputUN = ({ target: { value } }) => setUserName(value);
+    const onInputPW = ({ target: { value } }) => setPassword(value);
+    const onInputMN = ({ target: { value } }) => setManagerName(value);
+    const onInputRN = ({ target: { value } }) => setRoleName(value);
+    const onInputALE = ({ target: { value } }) => setAnnualLeaveEntitlement(value);
+    const onInputMLE = ({ target: { value } }) => setMedicalLeaveEntitlement(value);
+    const onInputCLE = ({ target: { value } }) => setCompensationLeaveEntitlement(value);
+
+    const [validated, setValidated] = useState(false);
+
+    const formRef = useRef();
+
+    useEffect(() => {
+        setEditDataOnLoad(id, formRef);
+      }, []);
+
+    const onFormSubmit = (event) => {
+        event.preventDefault();
+        const form = event.currentTarget;
+        console.log(id);
+        if (form.checkValidity() === false) {
+            event.stopPropagation();
+            setValidated(true);
+        } else {
+            editEmployeeInfo({
+                userName: formRef.current.querySelector("#formUserName").value,
+                password: formRef.current.querySelector("#formPassword").value,
+                managerName: formRef.current.querySelector("#formManager").value,
+                roleName: formRef.current.querySelector("#formRole").value,
+                annualLeaveEntitlement: formRef.current.querySelector("#formAnnual").value,
+                medicalLeaveEntitlement: formRef.current.querySelector("#formMedical").value,
+                compensationLeaveEntitlement: formRef.current.querySelector("#formCompensation").value,
+                id: id
+
+            }).then((response) => {
+                if (response.status == 200) {
+                    console.log(JSON.stringify(response.data));
+                    setAlertMsg(JSON.stringify(response.data));
+                    setShowAlert(true);
+                }
+            }).catch((error) => {
+                setAlertMsg("error, please try again.");
+                setShowAlert(true);
+            });
+        }
+    };
+
+    return (
+        <LoginCheckWrapper>
```

Comment:
You can update <LoginCheckWrapper> with the latest props to check for roles & run data loading methods. Check [this line](https://github.com/Qibaoooo/LAPS/blob/main/frontend/src/pages/managerClaimHistory.js#L29) for example :D